core/src/main/java/bisq/core/offer/OpenOfferManager.java

```diff
@@ -1143,7 +1178,33 @@ private boolean isBsqSwapOfferLackingFunds(OpenOffer openOffer) {
                 openOffer.isBsqSwapOfferHasMissingFunds();
     }
 
+    public boolean isSpam(OpenOffer newOffer) {
+        // an offer is spam if the user has another open offer on the same ccy, payment method, and reserved UTXO
+        long matchingOffers = openOffers.stream()
+                .filter((openOffer -> !openOffer.getOffer().isBsqSwapOffer()))
+                .filter(openOffer -> !openOffer.isDeactivated())
+                .filter(openOffer -> !openOffer.getShortId().equalsIgnoreCase(newOffer.getShortId()))
+                .filter(openOffer1 -> {
+                    Offer newOffer1 = newOffer.getOffer();
+                    Offer offer1 = openOffer1.getOffer();
+                    return
+                        offer1.getOfferFeePaymentTxId().equalsIgnoreCase(newOffer1.getOfferFeePaymentTxId()) &&
+                        offer1.getPaymentMethodId().equalsIgnoreCase(newOffer1.getPaymentMethodId()) &&
+                        offer1.getCounterCurrencyCode().equalsIgnoreCase(newOffer1.getCounterCurrencyCode()) &&
+                        offer1.getBaseCurrencyCode().equalsIgnoreCase(newOffer1.getBaseCurrencyCode());
+                })
+                .count();
+        if (matchingOffers > 0) {
+            log.info("{} is considered spam", newOffer.getShortId());
+        }
+        return matchingOffers > 0;
```

Comment:
A anyMatch would do it as well. The info log would be in the caller and can be a warning log.